zoo/repos/views.py

```diff
@@ -14,16 +24,61 @@ class RepoList(ListView):
 
 
 def repo_details(request, provider, repo_id):
-
-    if repo_id is None:
-        return JsonResponse({"error": "repository id not specified"}, status=400)
-    if provider is None:
-        return JsonResponse({"error": "provider is not specified"}, status=400)
-
     scm_module = get_scm_module(provider)
 
     try:
         details = scm_module.get_project_details(repo_id)
     except RepositoryNotFoundError:
         raise Http404(f"Project {repo_id} not found")
+
     return JsonResponse(details)
+
+
+def _parse_file(path, base=None):
+    try:
+        parser = ResolvingParser(str(path))
+        return parser.specification
+    except (
+        ValidationError,
+        ResolutionError,
+        ComposerError,
+        ScannerError,
+        AttributeError,
+    ) as err:
+        log.info(
+            "repos.views.openapi.invalid", path=str(path.relative_to(base)), error=err
+        )
+
+
+def openapi_definition(request, provider, repo_id):
+    try:
+        repository = Repository.objects.get(remote_id=repo_id, provider=provider)
+    except Repository.DoesNotExist:
+        return JsonResponse({"error": "repository not found"}, status=400)
```

Comment:
Doesn't HTTP status `404` make more sense here?

```suggestion
        return JsonResponse({"error": "repository not found"}, status=404)
```

Reply:
Actually yep, agree